libs/common-utils/src/displayTime.tsx

```diff
@@ -1,8 +1,31 @@
 import ms from 'ms'
 
 const [oneD, oneH, oneM, oneS] = [ms('1d'), ms('1h'), ms('1m'), ms('1s')]
-export function displayTime(time: number): string {
+
+/**
+ * Formats time in milliseconds into a human-readable string
+ * @param time Time in milliseconds
+ * @param expandedUnits If true, uses expanded unit names (e.g., "30 sec", "5 min", "2 hrs") instead of compact ones (e.g., "30s", "5m", "2h")
+ * @returns Formatted time string
+ */
+export function displayTime(time: number, expandedUnits = false): string {
   const timeMs = ms(`${time}ms`)
+
+  // For expanded units format, use a different approach based on the largest unit
+  if (expandedUnits) {
```

Comment:
The function still works wrongly, see example:
https://codesandbox.io/p/sandbox/lc77ls